File: packages/shared/src/utils/logger.ts
import pino from "pino";
import { ENVIRONMENT } from "../config/constants";

const logger = pino({
    level: ENVIRONMENT === "test" ? "silent" : ENVIRONMENT === "production" ? "info" : "debug",
    transport: {
        target: 'pino-pretty',
        options: {
            colorize: ENVIRONMENT === "development",
            sync: ENVIRONMENT === "test",
        }
    }

})

export default logger;

File: apps/video-worker/src/utils/logger.py
from pypino import PyPino
import os
ENVIRONMENT = os.getenv('ENVIRONMENT', 'development')

logger = PyPino(name="video-worker")

if not hasattr(logger, "warning") and hasattr(logger, "warn"):
    logger.warning = logger.warn

if ENVIRONMENT != 'production':
    logger.level("debug")
